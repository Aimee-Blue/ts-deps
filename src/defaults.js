// @ts-check
module.exports = Object.freeze({
  rootDir: 'src',
  outDir: 'lib',
  extensions: ['js', 'jsx', 'ts', 'tsx', 'json'],
  copyAdditional: [
    '**/*.d.ts',
    '**/*.json',
    '!**/__tests__/**',
    '!__integration-tests__/**',
  ],
  nodeVersion: '8.12',
  unitTestsGlob: 'src/**/__tests__/**',
  integrationTestsGlob: 'src/__integration-tests__/**',
  optionsFileName: 'ts-deps.config.js',
});
