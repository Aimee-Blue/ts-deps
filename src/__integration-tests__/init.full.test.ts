import {
  buildAndPack,
  ROOT,
  emptyDirSafe,
  spawnAndCheck,
  sortedDirectoryContents,
} from './helpers';
import { join } from 'path';
import { ensureDir, copy } from 'fs-extra';
import { defaults } from '../defaults';

jest.setTimeout(5 * 60 * 1000);

describe('init', () => {
  describe('given fresh directory after init', () => {
    const initDir = join(ROOT, 'integration-test-init-full');

    beforeAll(async () => {
      const { packageLocation } = await buildAndPack();

      await emptyDirSafe(initDir);
      await ensureDir(initDir);

      // to maky sure husky will not conflict with hooks of the main repo
      spawnAndCheck('git', ['init'], {
        cwd: initDir,
      });

      spawnAndCheck('npm', ['init', '-y'], {
        cwd: initDir,
      });

      const sourceLocation = join(ROOT, 'test-sources', 'init-full');

      await copy(sourceLocation, initDir, {
        overwrite: true,
      });

      spawnAndCheck('npm', ['add', 'file:' + packageLocation, '--save-dev'], {
        cwd: initDir,
      });

      spawnAndCheck('npx', ['ts-deps', 'init', '--force'], {
        cwd: initDir,
      });
    });

    it('should output correct files', async () => {
      const contents = await sortedDirectoryContents(initDir);

      expect(contents).toMatchSnapshot();
    });

    it('should build and test and lint', async () => {
      spawnAndCheck('npm', ['run', 'build'], {
        cwd: initDir,
      });

      spawnAndCheck('npm', ['run', 'declarations'], {
        cwd: initDir,
      });

      spawnAndCheck('npm', ['run', 'test'], {
        cwd: initDir,
      });

      spawnAndCheck('npm', ['run', 'type-check'], {
        cwd: initDir,
      });

      spawnAndCheck('npm', ['run', 'lint'], {
        cwd: initDir,
      });

      spawnAndCheck('npm', ['run', 'check'], {
        cwd: initDir,
      });

      spawnAndCheck('npm', ['run', 'release', '--version'], {
        cwd: initDir,
      });

      // tslint:disable-next-line
      const outDir = join(initDir, defaults.libOutDir);

      const libContents = await sortedDirectoryContents(outDir);

      expect(libContents).toMatchSnapshot();

      spawnAndCheck('npm', ['run', 'clean'], {
        cwd: initDir,
      });
    });

    describe('given changes to ts-deps config', () => {
      beforeAll(async () => {
        const sourceLocation = join(ROOT, 'test-sources', 'init-full-patch');

        await copy(sourceLocation, initDir, {
          overwrite: true,
        });

        spawnAndCheck('npx', ['ts-deps', 'patch'], {
          cwd: initDir,
        });
      });

      it('should patch', () => {
        return;
      });
    });
  });
});
