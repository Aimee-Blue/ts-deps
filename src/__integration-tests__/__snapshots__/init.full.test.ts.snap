// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init given fresh directory after init should build and test and lint 1`] = `
Array [
  "index.d.ts",
  "index.js",
  "max/",
  "max/index.d.ts",
  "max/index.js",
  "max/max.d.ts",
  "max/max.js",
  "min/",
  "min/index.d.ts",
  "min/index.js",
  "min/min.js",
  "other.d.ts",
  "other.js",
]
`;

exports[`init given fresh directory after init should output correct files 1`] = `
Array [
  ".gitignore",
  ".yarnrc",
  "babel.config.js",
  "commitlint.config.js",
  "jest.config.integration.js",
  "jest.config.js",
  "package-lock.json",
  "package.json",
  "prettier.config.js",
  "release.config.js",
  "scripts/",
  "scripts/build.js",
  "scripts/clean.js",
  "scripts/combineCoverage.js",
  "src/",
  "src/__integration-tests__/",
  "src/__tests__/",
  "src/__tests__/max.test.ts",
  "src/__tests__/min.test.js",
  "src/index.ts",
  "src/max/",
  "src/max/index.d.ts",
  "src/max/index.js",
  "src/max/max.ts",
  "src/min/",
  "src/min/index.d.ts",
  "src/min/index.js",
  "src/min/min.js",
  "src/other.d.ts",
  "src/other.js",
  "ts-deps.config.js",
  "tsconfig.declarations.json",
  "tsconfig.json",
  "tslint.json",
]
`;
